ledger/src/sigverify_shreds.rs

```diff
@@ -588,41 +590,46 @@ mod tests {
         solana_logger::setup();
         let mut batch = PinnedPacketBatch::default();
         let cache = RwLock::new(LruCache::new(/*capacity:*/ 128));
-        let mut shred = Shred::new_from_data(
-            slot,
-            0xc0de,
-            0xdead,
-            &[1, 2, 3, 4],
-            ShredFlags::LAST_SHRED_IN_SLOT,
+        let shredder = Shredder::new(slot, slot.saturating_sub(1), 0, 0).unwrap();
+        let keypair = Keypair::new();
+        let reed_solomon_cache = ReedSolomonCache::default();
+        let (shreds, _) = shredder.entries_to_merkle_shreds_for_tests(
+            &keypair,
+            &[],
+            true,
+            Some(Hash::default()),
             0,
             0,
-            0xc0de,
+            &reed_solomon_cache,
+            &mut ProcessShredsStats::default(),
         );
-        let keypair = Keypair::new();
-        shred.sign(&keypair);
-        batch.resize(1, Packet::default());
-        batch[0].buffer_mut()[..shred.payload().len()].copy_from_slice(shred.payload());
-        batch[0].meta_mut().size = shred.payload().len();
+        batch.resize(shreds.len(), Packet::default());
+        for i in 0..shreds.len() {
+            batch[i].buffer_mut()[..shreds[i].payload().len()].copy_from_slice(shreds[i].payload());
+            batch[i].meta_mut().size = shreds[i].payload().len();
+        }
         let batch = PacketBatch::from(batch);
         let mut batches = [batch];
 
         let leader_slots = HashMap::from([(slot, keypair.pubkey())]);
         let rv = verify_shreds_cpu(thread_pool, &batches, &leader_slots, &cache);
-        assert_eq!(rv, vec![vec![1]]);
+        assert_eq!(rv.into_iter().flatten().all_equal_value().unwrap(), 1);
 
         let wrong_keypair = Keypair::new();
         let leader_slots = HashMap::from([(slot, wrong_keypair.pubkey())]);
         let rv = verify_shreds_cpu(thread_pool, &batches, &leader_slots, &cache);
-        assert_eq!(rv, vec![vec![0]]);
+        assert_eq!(rv.into_iter().flatten().all_equal_value().unwrap(), 0);
 
         let leader_slots = HashMap::new();
         let rv = verify_shreds_cpu(thread_pool, &batches, &leader_slots, &cache);
-        assert_eq!(rv, vec![vec![0]]);
+        assert_eq!(rv.into_iter().flatten().all_equal_value().unwrap(), 0);
 
         let leader_slots = HashMap::from([(slot, keypair.pubkey())]);
-        batches[0].first_mut().unwrap().meta_mut().size = 0;
+        batches[0]
+            .iter_mut()
+            .for_each(|mut pr| pr.meta_mut().size = 0);
```

Comment:
nit: `pr` to abbreviate `packet` ? Can we just use `packet` ?

Reply:
Hmm ok, I'd still prefer `packet` or `packet_ref`; I think 1-2 letter variables names decrease readability and this one was not immediately obvious

As for `packet` vs `packet_ref`, the fact that it is a `PacketRef` and not a `&Packet` is an implementation detail that is not overly valuable to convey to the reader IMO.

Reply:
Renamed to packet_ref in https://github.com/anza-xyz/agave/pull/7171/commits/1ab6d08a24a682c35cf591f966170c622a943d00